cf/data/data.py

```diff
@@ -3389,12 +3370,19 @@ def _binary_operation(self, other, method):
             atol = self._atol
 
         # ------------------------------------------------------------
-        # Perform the binary operation with data0 (self) and data1 (other)
+        # Perform the binary operation with data0 (self) and data1
+        # (other)
         # ------------------------------------------------------------
         if method == "__eq__":
-            result = da.isclose(dx0, dx1, rtol=rtol, atol=atol)
+            if dx0.dtype.kind in "US" or dx1.dtype.kind in "US":
```

Comment:
Good spot and catch. Maybe we should update the test that covers the various binary operations, `test_Data_BINARY_AND_UNARY_OPERATORS`, to check some/more string type operations (not necessarily in this PR)? (I am already updating it a little to do some more complex unit checking to cover `_combined_units` (or raising an issue to register that this should be done at some point) as we discussed recently. Maybe I can do it as part of that update.)

Reply:
Doing as part of that update sounds good. I wavered about putting the fix in to _binary_operator in this PR, but decided to go for it as it was convenient, and might have been the only update to this area.

Reply:
All good, might as well include minor things in other PRs to avoid a long chain of PRs :)